pkg/cache/internal/informers.go

```diff
@@ -186,6 +187,12 @@ type Informers struct {
 // Start calls Run on each of the informers and sets started to true. Blocks on the context.
 // It doesn't return start because it can't return an error, and it's not a runnable directly.
 func (ip *Informers) Start(ctx context.Context) error {
+	select {
+	case <-ip.startWait:
```

Comment:
Do we need a lock around this?

Reply:
No, this is a channel

Reply:
That's not how I meant it :)

What if you have one goroutine in l.198 and another one then just going through l.191?
I would assume the same data race.

Don't we have to make sure an additional goroutine can't go through l.190-l.195 while another one is inside the critical section in l.197-216?

Reply:
Put all of it under the lock